test/ptp.go

```diff
@@ -0,0 +1,425 @@
+						Expect(podLogs).Should(ContainSubstring(profileSlave),
+							fmt.Sprintf("Profile \"%s\" not found in pod's log %s", profileSlave, pod.Name))
+					}
+					if podRole(pod, ptpGrandmasterNodeLabel) {
+						Expect(podLogs).Should(ContainSubstring(profileMaster),
+							fmt.Sprintf("Profile \"%s\" not found in pod's log %s", profileSlave, pod.Name))
+					}
+				}
+			})
+
+			// 25738
+			It("Slave can sync to master", func() {
+				var masterID string
+				var slaveMasterID string
+				grandMaster := "assuming the grand master role"
+				for _, pod := range ptpRunningPods {
+					if podRole(pod, ptpGrandmasterNodeLabel) {
+						podLogs, err := pods.GetLog(&pod)
+						Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+						Expect(podLogs).Should(ContainSubstring(grandMaster),
+							fmt.Sprintf("Log message \"%s\" not found in pod's log %s", grandMaster, pod.Name))
+						for _, line := range strings.Split(podLogs, "\n") {
+							if strings.Contains(line, "selected local clock") && strings.Contains(line, "as best master") {
+								masterID = strings.Split(line, " ")[4]
+							}
+						}
+					}
+					if podRole(pod, ptpSlaveNodeLabel) {
+						podLogs, err := pods.GetLog(&pod)
+						Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+						for _, line := range strings.Split(podLogs, "\n") {
+							if strings.Contains(line, "new foreign master") {
+								slaveMasterID = strings.Split(line, " ")[6]
+							}
+						}
+					}
+				}
+				Expect(masterID).NotTo(BeNil())
+				Expect(slaveMasterID).NotTo(BeNil())
+				Expect(slaveMasterID).Should(HavePrefix(masterID), "Error match MasterID with the SlaveID. Slave connected to another Master")
+			})
+		})
+	})
+
+	Describe("Test NodeName Selector", func() {
+		var testPtpPod v1core.Pod
+		var ptpConfigSlave ptpv1.PtpConfig
+
+		BeforeEach(func() {
+			ptpConfigName := "test"
+
+			response, err := client.Client.CoreV1Interface.RESTClient().Get().AbsPath(configPtpAPIPath + "/slave").DoRaw()
+			Expect(err).NotTo(HaveOccurred())
+			err = json.Unmarshal(response, &ptpConfigSlave)
+			Expect(err).NotTo(HaveOccurred())
+			nodes, err := client.Client.Nodes().List(metav1.ListOptions{
+				LabelSelector: ptpSlaveNodeLabel,
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(nodes.Items)).To(BeNumerically(">", 0),
+				fmt.Sprintf("PTP Nodes with label %s are not deployed on cluster", ptpSlaveNodeLabel))
+
+			ptpConfigTest := mutateProfile(ptpConfigSlave, ptpConfigName, nodes.Items[0].Name)
+
+			status := client.Client.CoreV1Interface.RESTClient().Post().AbsPath(configPtpAPIPath).
+				Resource("ptpconfigs").Body(ptpConfigTest).Context(context.TODO()).Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("PTP config creation Error"))
+
+			testPtpPod, err = getPtpPodOnNode(nodes.Items[0].Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			testPtpPod, err = getPtpPodOnNode(nodes.Items[0].Name)
```

Comment:
nit: This is executed twice, but it's good for testing :-)

Reply:
you are right thanks!